src/test/scala/org/broadinstitute/dsde/rawls/webservice/AdminApiServiceSpec.scala

```diff
@@ -429,4 +434,170 @@ class AdminApiServiceSpec extends FlatSpec with HttpService with ScalatestRouteT
         assertResult(StatusCodes.Forbidden) { status }
       }
     }
+
+
+  it should "return 201 when creating a new group" in withTestDataApiServices { services =>
+    val group = new RawlsGroupRef(RawlsGroupName("test_group"))
+
+    Post(s"/admin/groups", HttpEntity(ContentTypes.`application/json`, group.toJson.toString)) ~>
+      sealRoute(services.adminRoutes) ~>
+      check {
+        assertResult(StatusCodes.Created) { status }
+      }
+  }
+
+  it should "return 409 when trying to create a group that already exists" in withTestDataApiServices { services =>
+    val group = new RawlsGroupRef(RawlsGroupName("test_group"))
+
+    Post(s"/admin/groups", HttpEntity(ContentTypes.`application/json`, group.toJson.toString)) ~>
+      sealRoute(services.adminRoutes) ~>
+      check {
+        assertResult(StatusCodes.Created) { status }
+      }
+    Post(s"/admin/groups", HttpEntity(ContentTypes.`application/json`, group.toJson.toString)) ~>
+      sealRoute(services.adminRoutes) ~>
+      check {
+        assertResult(StatusCodes.Conflict) { status }
+      }
+  }
+
+  it should "return 200 when deleting a group that exists" in withTestDataApiServices { services =>
+    val group = new RawlsGroupRef(RawlsGroupName("test_group"))
+
+    Post(s"/admin/groups", HttpEntity(ContentTypes.`application/json`, group.toJson.toString)) ~>
+      sealRoute(services.adminRoutes) ~>
+      check {
+        assertResult(StatusCodes.Created) { status }
+      }
+    Delete(s"/admin/groups", HttpEntity(ContentTypes.`application/json`, group.toJson.toString)) ~>
+      sealRoute(services.adminRoutes) ~>
+      check {
+        assertResult(StatusCodes.OK) { status }
+      }
+  }
+
+  it should "return 404 when trying to delete a group that does not exist" in withTestDataApiServices { services =>
+    Delete(s"/admin/groups/doesnotexist") ~>
+      sealRoute(services.adminRoutes) ~>
+      check {
+        assertResult(StatusCodes.NotFound) { status }
+      }
+  }
+
+  it should "return 200 when listing the members of a group" in withTestDataApiServices { services =>
+    val group = new RawlsGroupRef(RawlsGroupName("test_group"))
+    val subGroup = new RawlsGroupRef(RawlsGroupName("dbgap"))
```

Comment:
Add a member to the subgroup to show that list members doesn't show it because it only goes down one level.